File: lib/py/msgpact/internal/types/VBoolean.py
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from ..validation.ValidateContext import ValidateContext
    from ...RandomGenerator import RandomGenerator
    from ..validation.ValidationFailure import ValidationFailure
    from .VTypeDeclaration import VTypeDeclaration
    from ..generation.GenerateContext import GenerateContext

from .VType import VType

_BOOLEAN_NAME: str = "Boolean"


class VBoolean(VType):

    def get_type_parameter_count(self) -> int:
        return 0

    def validate(self, value: object,
                 type_parameters: list['VTypeDeclaration'], ctx: 'ValidateContext') -> list['ValidationFailure']:
        from ..validation.ValidateBoolean import validate_boolean
        return validate_boolean(value)

    def generate_random_value(self, blueprint_value: object, use_blueprint_value: bool, type_parameters: list['VTypeDeclaration'], ctx: 'GenerateContext') -> object:
        from ..generation.GenerateRandomBoolean import generate_random_boolean
        return generate_random_boolean(blueprint_value, use_blueprint_value, ctx)

    def get_name(self) -> str:
        return _BOOLEAN_NAME

File: lib/java/src/main/java/io/github/msgpact/internal/types/VBoolean.java
package io.github.msgpact.internal.types;

import static io.github.msgpact.internal.generation.GenerateRandomBoolean.generateRandomBoolean;
import static io.github.msgpact.internal.validation.ValidateBoolean.validateBoolean;

import java.util.List;

import io.github.msgpact.internal.generation.GenerateContext;
import io.github.msgpact.internal.validation.ValidateContext;
import io.github.msgpact.internal.validation.ValidationFailure;

public class VBoolean implements VType {

    public static final String _BOOLEAN_NAME = "Boolean";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<VTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateBoolean(value);
    }

    @Override
    public Object generateRandomValue(Object blueprintValue, boolean useBlueprintValue,
            List<VTypeDeclaration> typeParameters, GenerateContext ctx) {
        return generateRandomBoolean(blueprintValue, useBlueprintValue, ctx);
    }

    @Override
    public String getName() {
        return _BOOLEAN_NAME;
    }

}
